sinonjs/sinon
# Prepare documentation using site in GitHub Pages

This is a PR for **master**, it implements a new documentation site using GitHub Pages (well Jekyll underneath).

This is the squashed version of my work-in-progress branch. I can't imagine anyone being interested in seeing all the detours I took to get here... if I am wrong, please speak up and I'll post a link to it.

I (now) think we can do everything that we need to with GitHub Pages, so there's really no reason to complicate things by having yet another system to manage.

---

There's still a lot of work left to do on the new documentation website, however I think we should merge this branch, as it will:

* allow me to finish configuring GitHub Pages, so we can see that the site actually works there
* allow others to contribute to finishing the documentation site

If this PR is merged, I will update https://github.com/sinonjs/sinon/issues/991 to track the remaining work. Specifically, I will (as a minimum) add these tasks.

- [x] Script for copying `docs/release-source/` into `docs/_releases/` as a new release, with the new release's id.
- [x] Script for setting `current_release` and/or `next_release` values in `docs/_config.yml` whenever a new release is created
- [x] Update Changelog.txt and AUTHORS whenever a new release is cut
- [x] Create a Jekyll collection for how to articles
- [ ] Create a Jekyll collection for packaged versions for download (http://sinonjs.org/download/)
- [ ] Script for copying the packaged release from `pkg/` to packaged versions, when a new release is cut
- [ ] Script for copying `Changelog.txt` into the documentation site whenever a new release is cut
- [ ] Link to the Changelog
- [ ] Add prominent links to source and issues on GitHub
- [ ] Go through the documentation of `v1.17.6` and remove any changes that have been made for `v2.0.0`. Perhaps we can re-import the current documentation from `sinonjs/sinon-docs`

I will try to convince the creator of the new favicon to create a better design and some illustrations for the site.


---

#### To verify

1. Check out this branch
1. `cd docs`
1. `gem install bundler`
1. `bundle install`
1. `bundle exec jekyll serve`
1. Navigate to http://localhost:4000/
1. Click around, and see if you can find any errors
